lib/deimos/active_record_consume/mass_updater.rb

```diff
@@ -84,15 +87,60 @@ def import_associations(record_list)
         end
       end
 
+      # rubocop:disable Metrics/AbcSize
+      # Upsert associated records prior to upserting primary records
+      # @param record_list [BatchRecordList]
+      def save_associations_first(record_list)
+        associations = Hash.new([])
+        record_list.associations.each do |assoc|
+          col = @bulk_import_id_column if assoc.klass.column_names.include?(@bulk_import_id_column)
+          associations[[assoc, col]] = []
+        end
```

Comment:
You should be able to remove lines 95-98.

Reply:
I would still need these lines to determine the `bulk_import_column` instead of checking for batch record

Reply:
Ah - didn't make the connection between this and line 100. -\_- Maybe rename `associations` to something else - it's mirroring `record.associations` but isn't actually recording the same thing.

In that case there's no need to use `Hash.new([])`.